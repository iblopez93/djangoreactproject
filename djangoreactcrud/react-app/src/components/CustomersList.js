import React, {Component} from 'react';
import CustomersService from './CustomersService';


const customersService = new CustomersService();

class CustomersList extends Component {


    constructor(props) {
        super(props);
        this.state = {customers: [],  prevPageURL: '',nextPageURL: ''};
        this.prevPage = this.prevPage.bind(this);
        this.nextPage = this.nextPage.bind(this);
        this.handleDelete = this.handleDelete.bind(this);
    }

    componentDidMount() {
        var self = this;
        customersService.getCustomers().then(function (result) {
            console.log(result);
            self.setState({customers: result.data, nextPageURL: result.nextlink,prevPageURL: result.prevlink})
        });
    }

    handleDelete(e, pk) {
        var self = this;
        customersService.deleteCustomer({pk: pk}).then(() => {
            var newArr = self.state.customers.filter(function (obj) {
                return obj.pk !== pk;
            });

            self.setState({customers: newArr})
        });
    }

    nextPage() {
        var self = this;
        console.log(this.state.nextPageURL);
        customersService.getCustomersByURL(this.state.nextPageURL).then((result) => {
            self.setState({customers: result.data, nextPageURL: result.nextlink})
        });
    }

    prevPage() {
        var self = this;
        console.log(this.state.prevPageURL);
        customersService.getCustomersByURL(this.state.prevPageURL).then((result) => {
            self.setState({customers: result.data, prevPageURL: result.prevlink})
        });
    }


    render() {
        return (<div className="customers--list">
                <table className="table">
                    <thead key="thead">
                    <tr>
                        <th>#</th>
                        <th>First Name</th>
                        <th>Last Name</th>
                        <th>Phone</th>
                        <th>Email</th>
                        <th>Address</th>
                        <th>Description</th>
                        <th>Actions</th>
                    </tr>
                    </thead>
                    <tbody>             {this.state.customers.map(c => <tr key={c.pk}>
                        <td>{c.pk}  </td>
                        <td>{c.first_name}</td>
                        <td>{c.last_name}</td>
                        <td>{c.phone}</td>
                        <td>{c.email}</td>
                        <td>{c.address}</td>
                        <td>{c.description}</td>
                        <td>
                            <button className="btn btn-danger" onClick={(e) => this.handleDelete(e, c.pk)}> Delete
                            </button>
                            <a className="btn btn-info" href={"/customer/" + c.pk}> Update
                            </a>
                        </td>
                    </tr>)}
                    </tbody>
                </table>
                <button className="btn btn-primary" onClick={this.prevPage}>Prev</button>
                <button className="btn btn-primary" onClick={this.nextPage}>Next</button>

            </div>
        );
    }
}

export default CustomersList;