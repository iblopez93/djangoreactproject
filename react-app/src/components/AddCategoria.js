// @flow
import React, { useState } from "react";
import PropTypes from 'prop-types';



export const AddCategoria = ({ setCategorias }) => {
    const [initialValue, setInitialValue] = useState('');
    const handleChange = (e) => {
        setInitialValue(e.target.value);
    };
    const handleSubmit = (e) => {
        e.preventDefault();
        if (initialValue.trim().length > 2) {
            setCategorias(categ => [initialValue, ...categ,]);
            setInitialValue('');
        }

    };
    return (
        <form onSubmit={handleSubmit}>
            <p>{initialValue}</p>
            <input
                type="text"
                value={initialValue}
                onChange={handleChange}
            />
        </form>
    );
};

AddCategoria.propTypes = {
    setCategorias: PropTypes.func.isRequired
}