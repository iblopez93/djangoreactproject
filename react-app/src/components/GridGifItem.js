import React from 'react'
import PropTypes from 'prop-types'


export const GridGifItem = ({ url, id, name, phone }) => {

    return (
        <div className="card animated fadeIn">
            <img
                src={url}
                alt={name}
            />
            <p>{name}</p>

        </div>
    )
}

GridGifItem.propTypes = {
    name: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}
