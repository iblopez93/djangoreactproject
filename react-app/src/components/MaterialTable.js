import React, {useState, useEffect} from 'react';

import {makeStyles} from '@material-ui/core/styles';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import {TablePagination} from "@material-ui/core";
import Paper from '@material-ui/core/Paper';
import CircularProgress from '@material-ui/core/CircularProgress';
import CustomersService from "./CustomersService";


const customersServices = new CustomersService();


const fetchData = () => customersServices.getCustomers();
// new Promise(resolve => {
//   const items = [
//     {
//       id: 1,
//       name: 'First Item',
//       created: new Date(),
//       high: 2935,
//       low: 1924,
//       average: 2429.5
//     },
//     {
//       id: 2,
//       name: 'Second Item',
//       created: new Date(),
//       high: 439,
//       low: 231,
//       average: 335
//     },
//     {
//       id: 3,
//       name: 'Third Item',
//       created: new Date(),
//       high: 8239,
//       low: 5629,
//       average: 6934
//     },
//     {
//       id: 4,
//       name: 'Fourth Item',
//       created: new Date(),
//       high: 3203,
//       low: 3127,
//       average: 3165
//     },
//     {
//       id: 5,
//       name: 'Fifth Item',
//       created: new Date(),
//       high: 981,
//       low: 879,
//       average: 930
//     }
//   ];
//
//   setTimeout(() => resolve(items), 1000);
// });
console.log(fetchData());
const usePaperStyles = makeStyles(theme => ({
    root: {margin: theme.spacing(2), textAlign: 'center'}
}));

const useProgressStyles = makeStyles(theme => ({
    progress: {margin: theme.spacing(2)}
}));

function MaybeLoading({loading}) {
    const classes = useProgressStyles();
    return loading ? (
        <CircularProgress className={classes.progress}/>
    ) : null;
}

export default function StatefulTables() {
    const classes = usePaperStyles();

    const [items, setItems] = useState([]);
    const [count, setCount] = useState(0);
    const [page, setPage] = useState(1);
    // const [prevPage, setPrevPage] = useState("");
    // const [nextPage, setNextPage] = useState("");
    const [pageSize, setPageSize] = useState(3);
    const [pageSizes] = useState([3, 6, 9]);
    const [loading, setLoading] = useState(true);

    useEffect(() => {
        fetchData().then(result => {
            setItems(result.data);
            setLoading(false);
            // setPrevPage(result.nextlink);
            // setNextPage(result.prevlink);
            setCount(result.count);
            setPageSize(result.pageSize);
            setPage(result.page);
        });
    }, []);

    // const handleNextPage = () => {
    //     var self = this;
    //     console.log(this.state.nextPageURL);
    //     customersServices.getCustomersByURL(this.state.nextPageURL).then((result) => {
    //         self.setState({customers: result.data, nextPageURL: result.nextlink})
    //     });
    // };
    //
    // const handlePrevPage = () => {
    //     var self = this;
    //     console.log(this.state.prevPageURL);
    //     customersServices.getCustomersByURL(this.state.prevPageURL).then((result) => {
    //         self.setState({customers: result.data, prevPageURL: result.prevlink})
    //     });
    // };
    const handlePageChange = (event, page) => {
        console.log(page+1);
        customersServices.getCustomersByURL(`?page=${page+1}&pageSize=${pageSize}`).then((result) => {
            setItems(result.data);
            setLoading(false);
            // setPrevPage(result.nextlink);
            // setNextPage(result.prevlink);
            setPageSize(result.pageSize);
            setCount(result.count);
            setPage(result.page);
        });
    };

    const handlePageSizeChange = (event) => {
        console.log(event);
        customersServices.getCustomersByURL(`?pageSize=${event.target.value}`).then((result) => {
            setItems(result.data);
            setLoading(false);
            // setPrevPage(result.nextlink);
            // setNextPage(result.prevlink);
            setPageSize(result.pageSize);
            setCount(result.count);
            setPage(result.page);
        });
    };

    return (
        <Paper className={classes.root}>
            <Table>
                <TableHead>
                    <TableRow>
                        <TableCell>First Name</TableCell>
                        <TableCell>Last Name</TableCell>
                        <TableCell align="right">Phone</TableCell>
                        <TableCell align="right">E-mail</TableCell>
                        <TableCell align="right">Address</TableCell>
                        <TableCell align="right">Description</TableCell>
                    </TableRow>
                </TableHead>
                <TableBody>
                    {items.map(item => {
                        return (
                            <TableRow key={item.pk}>
                                <TableCell component="th" scope="row">
                                    {item.first_name}
                                </TableCell>
                                <TableCell>{item.last_name}</TableCell>
                                <TableCell align="right">{item.phone}</TableCell>
                                <TableCell align="right">{item.email}</TableCell>
                                <TableCell align="right">{item.address}</TableCell>
                                <TableCell align="right">{item.description}</TableCell>

                            </TableRow>

                        );
                    })}
                </TableBody>
                <TablePagination count={count}
                                 page={page}
                                 onChangePage={handlePageChange}
                                 rowsPerPageOptions={pageSizes}
                                 onChangeRowsPerPage={handlePageSizeChange}
                                 rowsPerPage={pageSize}
                />
            </Table>
            <MaybeLoading loading={loading}/>
        </Paper>
    );
}
