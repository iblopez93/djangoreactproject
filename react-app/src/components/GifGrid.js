import React from 'react'
import { useFetchGifs } from '../hoooks/useFetchGifs';
import { GridGifItem } from './GridGifItem';
import PropTypes from 'prop-types';



export const GifGrid = ({ categorias }) => {
    const { data: imgs, loading } = useFetchGifs(categorias);


    return (
        <>
            <h3>{categorias}</h3>
            <div className="card-grid animated fadeIn">
                {loading && <p className="animated flash">'Cargando datos ...'</p>}


                {imgs.map((img) => (
                    <GridGifItem
                        key={img.id}
                        {...img}
                    />
                ))}

            </div>
        </>
    )
}

GifGrid.propTypes = {
    categorias: PropTypes.string.isRequired
}