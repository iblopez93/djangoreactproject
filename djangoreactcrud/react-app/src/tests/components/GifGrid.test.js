import React from 'react'
import { shallow } from "enzyme"
import { GifGrid } from "../../components/GifGrid"
import { useFetchGifs } from '../../hoooks/useFetchGifs';
jest.mock('../../hoooks/useFetchGifs');

const text = "AAA"
describe('Pruebas en el componente GifGrid', () => {
    test('Debe mostrarse el componente ', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        const wrapper = shallow(<GifGrid categorias={text} />)
        expect(wrapper).toMatchSnapshot();
    })
    test('Debe de mostrar el componente con datos de useFetchGifs ', () => {

        const gifs = [{
            id: 1,
            name: 'Julio Agoste Jimenez',
            email: 'iblopez@gmail.com',
            phone: '5452635446',
            url: 'http://127.0.0.1:3000/logo192.png'
        }];
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        const wrapper = shallow(<GifGrid categorias={text + ''} />)
        expect(wrapper).toMatchSnapshot();
    })

})
