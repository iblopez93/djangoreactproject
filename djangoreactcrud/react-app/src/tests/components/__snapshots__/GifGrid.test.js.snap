// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GifGrid Debe de mostrar el componente con datos de useFetchGifs  1`] = `
<Fragment>
  <h3>
    AAA
  </h3>
  <div
    className="card-grid animated fadeIn"
  >
    <GridGifItem
      email="iblopez@gmail.com"
      id={1}
      key="1"
      name="Julio Agoste Jimenez"
      phone="5452635446"
      url="http://127.0.0.1:3000/logo192.png"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en el componente GifGrid Debe mostrarse el componente  1`] = `
<Fragment>
  <h3>
    AAA
  </h3>
  <div
    className="card-grid animated fadeIn"
  >
    <p
      className="animated flash"
    >
      'Cargando datos ...'
    </p>
  </div>
</Fragment>
`;
